### PR TITLE
zh-cn: revise translation of 'your own automation environment'

### PR DESCRIPTION
**Pull request by hanyujie2002**
Relates to #19921

### CODE REVIEW

**Comment by hanyujie2002** [comment]
@jasonren0403 Would you please help me?

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md [hunk 1]
```diff
@@ -98,82 +100,66 @@ OK，现在我们来做一个快速的测试来验证一下一切是否正常。
 2. 将下面代码复制到文件中保存：
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md:100]:
```suggestion
2. 将以下内容复制到上述文件中，然后保存：
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md [hunk 2]
```diff
@@ -98,82 +100,66 @@ OK，现在我们来做一个快速的测试来验证一下一切是否正常。
...
 
-你会看到 Firefox 自动打开了一个窗口！Google 自动加载到标签页中，“webdriver”被输入到搜索框，然后搜索按钮被点击。然后，WebDriver 等待 2 秒，然后获取文本标题，如果标题是“webdriver - Google Search”, 将返回测试成功的消息。然后 WebDriver 关闭 Firefox 窗口并结束。
+你会看到 Firefox 自动打开一个窗口！Google 会自动加载到标签页中，并在搜索框中输入“webdriver”，然后点击搜索按钮。接着，WebDriver 会等待 1 秒钟，然后获取页面标题。如果标题是“webdriver - Google Search”，我们将返回一条消息，表示测试通过。然后我们再等待四秒钟，之后 WebDriver 会关闭 Firefox 窗口并停止。
 
 ## 一次测试多个浏览器
 
-接下来，让我们来试一下同时在多个浏览器上进行测试。这也是你经常碰到的情况！
+接下来，让我们来试一下同时在多个浏览器上进行测试。
 
 1. 在你的工程目录下创建另外一个新文件 `google_test_multiple.js`。你可以根据实际需要测试的浏览器情况，对我们添加的浏览器进行修改或删除等操作。但确保系统安装了正确的浏览器驱动。关于如何填写 `.forBrowser()` 方法中对浏览器描述的字符串，请参考 [Browser enum](https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/index_exports_Browser.html)。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md:133]:
```suggestion
1. 在你的工程目录下创建另外一个新文件 `google_test_multiple.js`。你可以根据实际需要测试的浏览器情况，对我们添加的浏览器进行修改或删除等操作。但确保系统安装了正确的浏览器驱动。关于如何填写 `.forBrowser()` 方法中对浏览器描述的字符串，请参考 [Browser 枚举值](https://www.selenium.dev/selenium/docs/api/javascript/global.html#Browser)页面。
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md [hunk 3]
```diff
@@ -98,82 +100,66 @@ OK，现在我们来做一个快速的测试来验证一下一切是否正常。
...
+       await driver.sleep(2000); // 延迟足够长的时间以看到搜索页面！
+       await driver.quit();
+     }
+   })();
    ```
 
+   > [!NOTE]
+   > 此函数是一个 {{glossary("IIFE")}}（立即调用函数表达式）。
+
 3. 在终端上，确保在你的工程目录下输入如下命令：
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md:121]:
```suggestion
3. 在终端上，确保在你的项目目录下输入如下命令：
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md [hunk 4]
```diff
@@ -98,82 +100,66 @@ OK，现在我们来做一个快速的测试来验证一下一切是否正常。
...
-         } else {
-           console.log("Test failed");
-         }
-         driver.quit();
-       });
-     });
-   }
    ```
 
 3. 在终端上，确保在你的工程目录下输入如下命令：
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md:165]:
```suggestion
3. 在终端上，确保在你的项目目录下输入如下命令：
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md [hunk 5]
```diff
@@ -182,52 +168,50 @@ OK，现在我们来做一个快速的测试来验证一下一切是否正常。
    node google_test_multiple
    ```
 
-4. 如果你用的是 Mac，并决定测试 Safari 浏览器，可能会产生一个错误信息：“Could not create a session: You must enable the 'Allow Remote Automation' option in Safari's Develop menu to control Safari via WebDriver.”如果是这样，根据指示重新尝试一遍。
+4. 如果你用的是 Mac，并决定测试 Safari 浏览器，可能会产生一个错误信息：“Could not create a session: You must enable the 'Allow Remote Automation' option in Safari's Develop menu to control Safari via WebDriver.”（无法创建会话：你必须启用 Safari 的 "开发 "菜单中的 "允许远程自动化 "选项，才能通过 WebDriver 控制 Safari）。如果是这样，请根据指示操作并重新尝试一遍。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md:171]:
```suggestion
4. 如果你用的是 Mac，并决定测试 Safari 浏览器，可能会产生一个错误信息：“Could not create a session: You must enable the 'Allow Remote Automation' option in Safari's Develop menu to control Safari via WebDriver.”（无法创建会话：你必须启用 Safari 的“开发”菜单中的“允许远程自动化”选项，才能通过 WebDriver 控制 Safari）。如果是这样，请根据指示操作并重新尝试一遍。
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md [hunk 6]
```diff
@@ -182,52 +168,50 @@ OK，现在我们来做一个快速的测试来验证一下一切是否正常。
...
 
-现在，我们像上次一样完成了测试，这一次浏览器的测试代码放到了 `searchTest()` 函数中。我们对多个浏览器创建了新的浏览器实例，然后将每一个实例传递给函数，这样就可以在 3 个浏览器下执行测试！
+现在，我们像上次一样完成了测试，这一次浏览器的测试代码放到了 `searchTest()` 函数中。我们对多个浏览器创建了新的浏览器实例，然后将每一个实例传递给函数，这样就可以在全部 3 个浏览器下执行测试！
 
 有意思吧？接下来我们继续，来了解一下 WebDriver 的语法基础。
 
 ## WebDriver 语法速成课程
 
 现在我们来看一下 webdriver 的一些关键语法。更完整的细节，可以参考 [selenium-webdriver JavaScript API 参考](https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/)以及 Selenium 主要的文档 [Selenium WebDriver](https://www.selenium.dev/documentation/webdriver/)，里面有用不同语言编写的丰富的学习示例。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md:179]:
```suggestion
现在我们来看一下 webdriver 的一些关键语法。更完整的细节，可以参考 [selenium-webdriver JavaScript API 参考](https://www.selenium.dev/selenium/docs/api/javascript/)以及 Selenium 主要的文档 [Selenium WebDriver](https://www.selenium.dev/documentation/webdriver/)，里面有用不同语言编写的丰富的学习示例。
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md [hunk 7]
```diff
@@ -239,7 +223,7 @@ driver.get("http://www.google.com");
 ```
 
 > [!NOTE]
-> 可以查看 [WebDriver class reference](https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/lib/webdriver_exports_WebDriver.html) 了解这一部分提到内容的详情。
+> 可以查看 [WebDriver class reference](https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/lib/webdriver_exports_WebDriver.html) 了解这一部分和下面提到的内容的详情。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md:226]:
```suggestion
> 可以查看 [WebDriver 类参考](https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/lib/webdriver_exports_WebDriver.html)了解这一部分和下面提到的内容的详情。
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md [hunk 8]
```diff
@@ -257,26 +241,41 @@ driver.get("http://localhost:8888/fake-div-buttons.html");
...
+(async function example() {
+  const driver = await new Builder().forBrowser(Browser.FIREFOX).build();
+  driver.get(
+    "https://mdn.github.io/learning-area/tools-testing/cross-browser-testing/accessibility/native-keyboard-accessibility.html",
+  );
+  const button = driver.findElement(By.css("button:nth-of-type(1)"));
+})();
 ```
 
 ### 测试元素
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md:281]: L283 should use localized link

https://www.selenium.dev/zh-cn/documentation/webdriver/elements/information/#%E6%96%87%E6%9C%AC%E5%86%85%E5%AE%B9

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md [hunk 9]
```diff
@@ -356,40 +422,71 @@ driver.sleep(2000).then(() => {
...
+  }
+})();
 ```
 
-WebDriver 等待 2 秒然后填充表单的文本框。接下来我们使用 `getAttribute()` 获取它的 `value` 属性值，并对它进行测试（如是否为空），最后将测试结果打印出来。
+WebDriver 会等待 2 秒然后填充表单的文本框。接下来我们使用 `getAttribute()` 获取它的 `value` 属性值，并对它进行测试（如是否为空），最后将测试结果打印出来。
 
 > [!NOTE]
-> 还有一个方法叫 [`wait()`](https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/lib/webdriver_exports_WebDriver.html#wait), 它是在一定的时间内对某个条件进行反复测试，然后再继续执行代码，它也使用了 [util 库](https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/lib/until.html)，其中定义了使用 `wait()` 的常见条件。
+> 还有一个方法叫 [`wait()`](https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/lib/webdriver_exports_WebDriver.html#wait)，它是在一定的时间内对某个条件进行反复测试，然后再继续执行代码，它也使用了 [util 库](https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/lib/until.html)，其中定义了使用 `wait()` 的常见条件。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md:472]:
```suggestion
> 还有一个方法叫 [`wait()`](https://www.selenium.dev/selenium/docs/api/javascript/WebDriver.html#wait)，它是在一定的时间内对某个条件进行反复测试，然后再继续执行代码，它也使用了 [util 库](https://www.selenium.dev/selenium/docs/api/javascript/lib_until.js.html)，其中定义了使用 `wait()` 的常见条件。
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md [hunk 10]
```diff
@@ -356,40 +422,71 @@ driver.sleep(2000).then(() => {
...
-当你运行它时，应该看到测试在执行，并且在测试完成后浏览器实例关闭。这对于不使你的计算机被大量的浏览器实例弄得杂乱无章是很有用的，特别是浏览器实例太多，导致计算机速度减慢。
+运行该程序时，你应该会看到测试正在执行，并且在测试完成后，浏览器实例会自动关闭。这有助于防止计算机因过多的浏览器实例而变得杂乱无章，特别是在浏览器实例过多导致计算机速度减慢的情况下。
 
 ## 测试最佳实践
 
 有很多编写最佳测试的实践方法，你可以参考[测试设计考虑](https://www.selenium.dev/zh-cn/documentation/test_practices/)来了解一些背景。总的来说，测试应该遵循如下几点：
 
 1. 使用好的定位策略：当你[同文档交互](#同文档交互)时，确保你使用的定位器和页面对象是不变的。如对元素进行测试，确保这个元素有固定的 ID 或页面位置，这样可以通过 CSS 选择器定位到它，不至于在下一个迭代就发生变化。尽可能让你的测试稳健，而不是有一点改动就会崩溃。
-2. 写原子测试：一个测试只测一件事，这样有利于跟踪哪一个测试文件执行的是哪种测试。举例来说，前面的`google_test.js` 就只测试了一个简单的用例——页面的搜索结果标题是否正确。我们可以给它改一个名字，这样当我们添加更多的测试文件时它的作用会更直观，比如改为 `results_page_title_set_correctly.js`。
+2. 写原子测试：一个测试只测一件事，这样有利于跟踪哪一个测试文件执行的是哪种测试。举例来说，前面的`google_test.js` 就只测试了一个简单的用例——页面的搜索结果标题是否正确。我们可以给它改一个名字，比如改为 `results_page_title_set_correctly.js`，这样当我们添加更多的测试文件时它的作用会更直观。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md:489]:
```suggestion
2. 写原子测试：一个测试只测一件事，这样有利于跟踪哪一个测试文件执行的是哪种测试。举例来说，前面的 `google_test.js` 就只测试了一个简单的用例——页面的搜索结果标题是否正确。我们可以给它改一个名字，比如改为 `results_page_title_set_correctly.js`，这样当我们添加更多的测试文件时它的作用会更直观。
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md [hunk 11]
```diff
@@ -409,79 +506,80 @@ driver.quit();
...
-               driver.quit();
-             }, 5000);
-           });
-         });
-     });
-   }
-
-   searchTextOnGoogle();
-   ```
+```js
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md:524]: note the indentation

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md [hunk 12]
```diff
@@ -503,113 +600,101 @@ driver.quit();
...
-3. 在 [BrowserStack automation dashboard](https://www.browserstack.com/automate) 中获取你的用户名和访问密钥（查看 _Username and Access Keys_），替换代码中 `YOUR-USER-NAME` 和 `YOUR-ACCESS-KEY` 的对应值（并确保它们安全存放）。
+3. 从 [BrowserStack Account - Summary](https://www.browserstack.com/accounts/profile/details) 获取用户名和访问密钥（请参阅*用户名和访问密钥*）。用实际用户名和访问密钥值替换代码中的 `YOUR-USER-NAME` 和 `YOUR-ACCESS-KEY` 占位符（并确保它们不泄露）。
 4. 执行下面命令：
 
    ```bash
    node bstack_google_test
    ```
 
-   测试被发送给 BrowserStackces, 测试结果会返回到你的控制台。这体现了包含报告机制的重要性！
+   测试被发送给 BrowserStackces，测试结果会返回到你的控制台。这体现了包含报告机制的重要性！
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md:660]:
```suggestion
   测试被发送给 BrowserStack，测试结果会返回到你的控制台。这体现了包含报告机制的重要性！
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md [hunk 13]
```diff
@@ -503,113 +600,101 @@ driver.quit();
...
 
-2. 现在更新一下 `capabilities` 对象，添加工程名——在右大括号前添加下面代码，记得要在上一行末增加一个逗号（在 BrowserStack 自动化仪表盘上，你可以修改 build 和 project 名称来组织不同窗口下的测试）：
+3. 现在更新一下 `capabilities` 对象，添加工程名——在右大括号前添加下面代码，记得要在上一行末增加一个逗号（在 BrowserStack 自动化仪表盘上，你可以修改 build 和 project 名称来组织不同窗口下的测试）：
 
    ```js
    'project' : 'Google test 2'
    ```
 
-3. 接下来获取当前会话的 `sessionId` , 就知道往哪儿发送请求（后面你会看到，ID 包含在请求的 URL 中）。将下面代码添加到创建 `driver` 对象（`var driver ...`）的代码块下面：
+4. 接下来获取当前会话的 `sessionId` ，从而知道往哪儿发送请求（后面你会看到，ID 包含在请求的 URL 中）。将下面代码添加到创建 `driver` 对象（`let driver ...`）的代码块下面：
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md:697]:
```suggestion
4. 接下来获取当前会话的 `sessionId`，从而知道往哪儿发送请求（后面你会看到，ID 包含在请求的 URL 中）。将下面代码添加到创建 `driver` 对象（`let driver …`）的代码块下面：
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md [hunk 14]
```diff
@@ -635,7 +720,7 @@ driver.quit();
            },
          });
        } else {
-         console.log("Test failed");
+         console.log("测试失败");
          request({
            uri: `https://YOUR-USER-NAME:YOUR-ACCESS-KEY@www.browserstack.com/automate/sessions/${sessionId}.json`,
            method: "PUT",
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md:726]: Line 739 - wrong link(should be https://live.browserstack.com/dashboard)
Line 745 - should be "遵循...模式"

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md [hunk 15]
```diff
@@ -791,10 +874,10 @@ driver.quit();
 
 ### 自己的远程服务器
 
-如果你不想使用 Sauce Labs 或 BrowserStack 之类的服务，你也可以设置自己的远程测试服务器。具体操作如下：
+如果你不想使用 Sauce Labs 或 BrowserStack 之类的服务，你也可以配置自己的远程测试服务器。具体操作如下：
 
 1. Selenium 的远端服务器要求运行 Java，从 [Java SE 下载页面](https://www.oracle.com/java/technologies/downloads/)下载适合你平台的最新 JDK 并安装。
-2. 接着，下载最新的 [Selenium 单机服务器](http://selenium-release.storage.googleapis.com/index.html)，它作为你的脚本和浏览器驱动之间的一个代理。选择最新的稳定版本（最好不要选 beta 版本），从列表中选择以“selenium-server-standalone”开头的文件。下载完成后，放到一个靠谱的地方，比如家目录下。如果你还没有将位置添加到 `PATH`，现在就需要添加了（查看[在 Node 中创建 Selenium](#在_node_下建立_selenium) 小节）。
+2. 接着，下载最新的 [Selenium 单机服务器](http://selenium-release.storage.googleapis.com/index.html)，它作为你的脚本和浏览器驱动之间的一个代理。选择最新的稳定版本（最好不要选 beta 版本），从列表中选择以“selenium-server-standalone”开头的文件。下载完成后，放到一个靠谱的地方，比如主目录下。如果你还没有将位置添加到 `PATH`，现在就需要添加了（查看[在 Node 中创建 Selenium](#在_node_下建立_selenium) 小节）。
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md:880]:
```suggestion
2. 接着，下载最新的 [Selenium 单机服务器](http://selenium-release.storage.googleapis.com/index.html)，它作为你的脚本和浏览器驱动之间的一个代理。选择最新的稳定版本（最好不要选 beta 版本），从列表中选择以“selenium-server-standalone”开头的文件。下载完成后，放到一个靠谱的地方，比如主目录下。如果你还没有将位置添加到 `PATH`，现在就需要添加了（查看[在 Node 中搭建 Selenium 环境](#在_node_下搭建_selenium_环境)小节）。
```

File: files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md [hunk 16]
```diff
@@ -791,10 +874,10 @@ driver.quit();
...
 3. 在作为服务器的电脑终端上执行如下代码，安装单机版服务器
```
Comment (jasonren0403) [inline files/zh-cn/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md:881]:
```suggestion
3. 在作为服务器的电脑终端上执行如下代码，安装独立服务器。
```

**Review by jasonren0403 — COMMENTED** [review]
sorry for late review, here are some suggestions

**Review by jasonren0403 — APPROVED** [review]
👍 thx

**Pull request merged**